tapdb/sqlc/queries/addrs.sql

```diff
@@ -1,9 +1,45 @@
--- name: InsertAddr :one
+-- name: UpsertAddr :one
 INSERT INTO addrs (
-    version, asset_version, genesis_asset_id, group_key, script_key_id,
-    taproot_key_id, tapscript_sibling, taproot_output_key, amount, asset_type,
-    creation_time, proof_courier_addr
-) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12) RETURNING id;
+    version,
+    asset_version,
+    genesis_asset_id,
+    group_key,
+    script_key_id,
+    taproot_key_id,
+    tapscript_sibling,
+    taproot_output_key,
+    amount,
+    asset_type,
+    creation_time,
+    proof_courier_addr
+) VALUES (
+    $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12
+) 
+ON CONFLICT (taproot_output_key) DO UPDATE
+SET
+    -- If the WHERE clause below is true (exact match on all other fields,
+    -- except for creation_time), we set taproot_output_key to its current
+    -- conflicting value. This is a no-op in terms of data change but allows
+    -- RETURNING id to work on the existing row.
+    taproot_output_key = excluded.taproot_output_key
+WHERE 
+    addrs.version = excluded.version
```

Comment:
Hmm, interesting approach. 

So this now returns `unable to insert addr: sql: no rows in result set` if some fields mismatch. Sounds good to me, but I think we should catch that on the RPC server and return a more user friendly error. Otherwise it's hard to understand what's wrong.

Reply:
ptal